### PR TITLE
New Resource: `azurerm_automation_connection_type`

### PR DESCRIPTION
**Pull request by wuxu92**
add a new resource for automation connection type.  

--- PASS: TestAccAutomationConnectionType_basic (131.51s)

### CODE REVIEW

File: website/docs/r/automation_connection_type.html.markdown [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
+subcategory: "Automation"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_automation_connection_type"
+description: |-
+  Manages a Automation.
```
Comment (katbyte) [inline website/docs/r/automation_connection_type.html.markdown:6]: this is incomplete?

File: internal/services/automation/automation_connection_type_resource.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+			ValidateFunc: validate.ConnectionTypeName,
+		},
+
+		"is_global": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+		},
+
+		"field_definitions": {
```
Comment (katbyte) [inline internal/services/automation/automation_connection_type_resource.go:62]: definitions feels redundant?
```suggestion
		"field": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by wuxu92** [comment]
@katbyte i have fixed the comments, but the CI failed because the terrafmt's new version breaks the process

```
go: github.com/katbyte/terrafmt@latest (in github.com/katbyte/terrafmt@v0.5.0):
	The go.mod file for the module providing named packages contains one or
	more exclude directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.
make: *** [GNUmakefile:17: tools] Error 1
Error: Process completed with exit code 2.
```

File: website/docs/r/automation_connection_type.html.markdown [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+subcategory: "Automation"
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_automation_connection_type"
+description: |-
+  Manages a Automation Connection Type.
+---
+
+# azurerm_automation_connection_type
+
+Manages a Automation Connection Type.
```
Comment (katbyte) [inline website/docs/r/automation_connection_type.html.markdown:11]: grammer
```suggestion
  Manages an Automation Connection Type.
---

# azurerm_automation_connection_type

Manages anAutomation Connection Type.
```

**Review by katbyte — APPROVED** [review]
LGTM aside from my comment 📜

**Pull request merged**